frontend/src/components/PasswordFormField.js

```diff
@@ -0,0 +1,70 @@
+import React, { useState, useEffect } from "react";
+import TextFormField from "./TextFormField";
+
+const PasswordFormField = ({ valid, onFieldChange, step }) => {
+  const [errorMessage, setErrorMessage] = useState({
+    password: "",
+    confirmPassword: "",
+  });
+  const [field, setField] = useState({ password: "", confirmPassword: "" });
+
+  useEffect(() => {
+    if (step !== 4) return;
+    const password = field.password.trim();
+    const confirmPassword = field.confirmPassword.trim();
+
+    if (confirmPassword.length > 0 && confirmPassword === password) {
+      setErrorMessage((errors) => ({
+        ...errors,
+        confirmPassword: "",
+      }));
+      valid(true);
+    } else if (confirmPassword.length === 0) {
+      setErrorMessage((errors) => ({ ...errors, confirmPassword: "" }));
+      valid(false);
+    } else {
+      setErrorMessage((errors) => ({
+        ...errors,
+        confirmPassword: "Les mots de passe ne sont pas identique",
+      }));
+      valid(false);
+    }
+  }, [field.password, field.confirmPassword, step, valid]);
```

Comment:
I'm a bit confused here, why are the error messages going inside the "confirmPassword" variable, the variable name is a bit confusing

Reply:
Error messages goes into confirmPassword as confirming a password implies that the "confirmPassword" field matches the "password" field, therefore the error is highlighted on the "confirmPassword" field.

Reply:
so you are replacing the content of the field with the error message ?

Reply:
No, replacing the error message underneath the field.

Reply:
so the variable has a special meaning in the context of setErrorMessage() ?